### PR TITLE
2 rework cambench cap fix algo

### PR DESCRIPTION
**Pull request by schlichtig**
Rework of CamBench_Cap

### CODE REVIEW

File: CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 public class CorrectedCrypto {
     public static void main(String[] args) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException {
-        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto.java:27]: 🔑  👍

File: CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto_meta2.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ api:
   javax.crypto.KeyGenerator
 
 description:
-  The KeyGenerator is used with an insecure algorithm (AES).
+  The KeyGenerator is used with a secure algorithm (AES).
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto_meta2.yaml:8]: 👍

File: CamBench_Cap/src/main/java/org/cambench/cap/basic/ecbmode/CorrectedEcbMode_meta.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ api:
   javax.crypto.Cipher
 
 description:
-  The AES algorithm is configured and initialized securely with the CBC mode of operation.
+  The AES algorithm is configured and initialized securely with the GCM mode of operation.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/basic/ecbmode/CorrectedEcbMode_meta.yaml:8]: 👍

File: CamBench_Cap/src/main/java/org/cambench/cap/fieldsensitivity/truepositive/brokencrypto/BrokenCrypto1.java [hunk 4]
```diff
@@ -26,7 +26,7 @@ public class BrokenCrypto1 {
     public static void main(String[] args) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException {
         CryptoClass cryptoClass = new CryptoClass();
         cryptoClass.cipher1 = "DES";
-        cryptoClass.cipher2 = "AES";
+        cryptoClass.cipher2 = "AES/GCM/NoPadding";
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/fieldsensitivity/truepositive/brokencrypto/BrokenCrypto1.java:29]: 👍  Good one!

File: CamBench_Cap/src/main/java/org/cambench/cap/interprocedural3/truepositive/brokencrypto/CorrectedCrypto.java [hunk 5]
```diff
@@ -35,7 +35,7 @@ public static void method1(String algo) throws NoSuchPaddingException, NoSuchAlg
     }
 
     public static void method2(String algorithm) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException {
-        String algoConfig = String.join("/", algorithm,"CBC","PKCS5Padding");
+        String algoConfig = String.join("/", algorithm,"GCM","PKCS5Padding");
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/interprocedural3/truepositive/brokencrypto/CorrectedCrypto.java]: What was your rationale to leave the PKCS5Padding for GCM?

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/falsepositive/ecbmode/EcbMode1_meta.yaml [hunk 6]
```diff
@@ -5,7 +5,7 @@ api:
   java.security.Cipher
 
 description:
-  A configuration for the AES cipher using the CBC mode is passed to a method in the reachable branch of a conditional 
+  A configuration for the AES cipher using the GCM mode is passed to a method in the reachable branch of a conditional
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/falsepositive/ecbmode/EcbMode1_meta.yaml:8]: Minor revision of the sentence. 
A configuration for the AES cipher using the GCM mode. 
This configuration is passed to a method in the reachable branch of a conditional statement, where it is used to initialize a cipher object.

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/falsepositive/brokencrypto/BrokenCrypto1_meta1.yaml [hunk 7]
```diff
@@ -5,7 +5,7 @@ api:
   javax.crypto.Cipher
 
 description:
-  A string containing the name of an secure cipher (AES) is passed to a in the reachable branch of a 
+  A string containing the name of a secure cipher (AES) is passed to a in the reachable branch of a
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/falsepositive/brokencrypto/BrokenCrypto1_meta1.yaml:8]: Correction of the sentence (article duplication): 
... is passed to the reachable branch of a conditional statement.

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 8]
```diff
@@ -26,7 +26,7 @@ crypto-usage:
 
 fix:
   description:
-    Replace the ECB mode of operation through another, secure mode, e.g. CBC mode.
+    Replace the ECB mode of operation through another, secure mode, e.g. GCM mode.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2path/truepositive/ecbmode/EcbMode1_meta.yaml:29]: mode, e.g., GCM mode.

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/objectpath/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 9]
```diff
@@ -26,7 +26,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/objectpath/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM.

like is used for comparisons and is not necessarily inclusive.

File: CamBench_Cap/src/main/java/org/cambench/cap/objectsensitivity/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 10]
```diff
@@ -26,7 +26,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/objectsensitivity/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM.

Why? See above

File: CamBench_Cap/src/main/java/org/cambench/cap/contextsensitivity/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 11]
```diff
@@ -24,7 +24,7 @@ crypto-usage:
 
 fix:
   description:
-    Replace the ECB mode in the configuration with another, more secure mode like CBC.
+    Replace the ECB mode in the configuration with another, more secure mode like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/contextsensitivity/truepositive/ecbmode/EcbMode1_meta.yaml:27]: , such as GCM.

Why? Explanaiton below in one of the comments.

File: CamBench_Cap/src/main/java/org/cambench/cap/interprocedural2/truepositive/brokencrypto/CorrectedCrypto.java [hunk 12]
```diff
@@ -29,7 +29,7 @@ public static void main(String[] args) throws NoSuchPaddingException, NoSuchAlgo
         method1(algo);
     }
     public static void method1(String algo) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException {
-        String algoConfig = String.join("/", algo,"CBC","PKCS5Padding");
+        String algoConfig = String.join("/", algo,"GCM","PKCS5Padding");
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/interprocedural2/truepositive/brokencrypto/CorrectedCrypto.java]: Replace PKCS5Padding with NoPadding

File: CamBench_Cap/src/main/java/org/cambench/cap/interprocedural2/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 13]
```diff
@@ -25,7 +25,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/interprocedural2/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM

Why? See a comment below.

File: CamBench_Cap/src/main/java/org/cambench/cap/interprocedural3/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 14]
```diff
@@ -25,7 +25,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/interprocedural3/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/fieldobject/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 15]
```diff
@@ -26,7 +26,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/fieldobject/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM.

File: CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2flow/truepositive/ecbmode/EcbMode1_meta.yaml [hunk 16]
```diff
@@ -26,7 +26,7 @@ crypto-usage:
 
 fix:
   description:
-    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like CBC.
+    Do not utilize the ECB mode of operation. Instead, use a more secure mode, like GCM.
```
Comment (akwick) [inline CamBench_Cap/src/main/java/org/cambench/cap/mixedsensitivities/inter2flow/truepositive/ecbmode/EcbMode1_meta.yaml]: , such as GCM

**Review by akwick — CHANGES_REQUESTED** [review]
Thanks for the extensive refactoring. I focused my review *only* on the changes.

I selected *Request changes* because of the two(?) cases where you use GCM with PKCS5Padding. Without them, I would have chosen *Approve* as these are more feedback and should not influence the evaluation results. ;)

File: CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto.java (hunk 1 above)
Comment (schlichtig) [reply CamBench_Cap/src/main/java/org/cambench/cap/basic/brokencrypto/CorrectedCrypto.java:27]: Good find !!!

**Comment by schlichtig** [comment]
Thank you @akwick 
I incorporated your findings!

**Review by akwick — APPROVED** [review]

**Comment by akwick** [comment]
@schlichtig thanks for changing!

**Pull request merged**